lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -38,6 +39,7 @@ import java.time.format.DateTimeParseException
 class SqlParser(private val ion: IonSystem) : Parser {
 
     private val trueValue: IonBool = ion.newBool(true)
+    private val datePatternRegex = Regex("\\d\\d\\d\\d-\\d\\d-\\d\\d")
```

Comment:
That's better I guess, but I really meant outside of the class too.  A singleton.